SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialStatics.cpp

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Utils/SpatialStatics.h"
+#include "GeneralProjectSettings.h"
+#include "SpatialConstants.h"
+#include "SpatialGDKSettings.h"
+#include "Utils/ActorGroupManager.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialStatics);
+
+bool USpatialStatics::SpatialNetworkingEnabled()
+{
+	return GetDefault<UGeneralProjectSettings>()->bSpatialNetworking;
+}
+
+bool USpatialStatics::SpatialOffloadingEnabled()
+{
+	return SpatialNetworkingEnabled() && GetDefault<USpatialGDKSettings>()->bEnableOffloading;
+}
+
+bool USpatialStatics::ActorGroupAssociationMatchesWorkerType(const AActor* Actor)
+{
+	if (Actor == nullptr)
+	{
+		return false;
+	}
+
+	UActorGroupManager* ActorGroupManager = UActorGroupManager::GetInstance();
+
+	if (UWorld* World = Actor->GetWorld())
+	{
+		FString WorkerType = World->GetGameInstance()->GetSpatialWorkerType();
+
+		FName ActorGroup = ActorGroupManager->GetActorGroupForClass(Actor->GetClass());
+		if (ActorGroup == NAME_None) {
+			UE_LOG(LogSpatialStatics, Error, TEXT("Could not find actor group for actor: %s"), *Actor->GetName());
+			return false;
+		}
+
+		FString ActorWorkerAssociation = ActorGroupManager->GetWorkerTypeForClass(Actor->GetClass());
+		if (ActorWorkerAssociation.Equals(TEXT(""))) {
+			UE_LOG(LogSpatialStatics, Error, TEXT("Could not find worker association for actor group %s"), *ActorGroup.ToString());
+			return false;
+		}
+
+		return ActorWorkerAssociation.Equals(WorkerType);
+	}
+
+	UE_LOG(LogSpatialStatics, Error, TEXT("World was nullptr in USpatialStatics::CanHaveAuthority"));
```

Comment:
Don't think this is a helpful error message if it happens in a blueprint. Can you put something that would help a blueprint user?

Reply:
The function was also renamed so this definitely needs to be updated.

Reply:
What kind of thing do you suggest for something that helps a blueprint user in this context @Vatyx ?

Reply:
Is there a case where the actor wouldn't have a valid world set when invoked from BP?

Reply:
A more helpful worker log could include the Actor name and the correct name of the function.

Reply:
I mean it had correct name of function when you came to reply, but strong comment regardless